corehq/apps/saved_reports/models.py

```diff
@@ -691,84 +695,141 @@ def _get_or_create_uuid(self):
-                )
+            report_text, excel_files = self._generate_report(attach_excel, title, emails)
+
+            # Both are empty if ALL the ReportConfigs in the ReportNotification
+            # have a start_date in the future (or an exception occurred)
+            if not report_text and not excel_files:
+                return
 
-                # Both are empty if ALL the ReportConfigs in the ReportNotification
-                # have a start_date in the future.
-                if not report_text and not excel_files:
-                    return
-
-                for email in emails:
-                    body = render_full_report_notification(None, report_text, email, self).content
-                    send_html_email_async(
-                        title, email, body,
-                        email_from=settings.DEFAULT_FROM_EMAIL,
-                        file_attachments=excel_files,
-                        smtp_exception_skip_list=LARGE_FILE_SIZE_ERROR_CODES)
+            self._send_emails(title, report_text, emails, excel_files)
+
+    @staticmethod
+    def _get_title(subject):
+        # only translate the default subject
+        return (
+            _(DEFAULT_REPORT_NOTIF_SUBJECT)
+            if subject == DEFAULT_REPORT_NOTIF_SUBJECT
+            else subject
+        )
+
+    def _generate_report(self, attach_excel, title, emails):
+        from corehq.apps.reports.views import get_scheduled_report_response
+        report_text = None
+        excel_files = None
+
+        try:
+            report_text, excel_files = get_scheduled_report_response(
+                self.owner, self.domain, self._id, attach_excel=attach_excel,
+                send_only_active=True
+            )
+        # TODO: Be more specific with our error-handling. If building the report could fail,
+        # we should have a reasonable idea of why
+        except Exception as er:
+            notify_exception(
+                None,
+                message="Encountered error while generating report",
+                details={
+                    'subject': title,
+                    'recipients': str(emails),
+                    'error': er,
+                }
+            )
+            if isinstance(er, ESError):
+                # An ElasticSearch error could indicate that the report itself is too large.
+                # Try exporting the report instead, as that builds the report in chunks,
+                # rather than all at once.
+                # TODO: narrow down this handling so that we don't try this if ElasticSearch is simply down,
+                # for example
+                self._export_report(emails, title)
+        return report_text, excel_files
+
+    def _send_emails(self, title, report_text, emails, excel_files):
+        from corehq.apps.reports.views import render_full_report_notification
+
+        email_is_too_large = False
+        total_sent = 0
+
+        for email in emails:
```

Comment:
nit: Use `enumerate` rather than maintaining a counter variable
```suggestion
        for total_sent, email in enumerate(emails):
```